opslevel/resource_opslevel_integration_google_cloud.go

```diff
@@ -0,0 +1,252 @@
+	var planModel integrationGoogleCloudResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	ownershipTagKeys, diags := SetValueToStringSlice(ctx, planModel.OwnershipTagKeys)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	input := opslevel.GoogleCloudIntegrationInput{
+		ClientEmail:           planModel.ClientEmail.ValueStringPointer(),
+		Name:                  planModel.Name.ValueStringPointer(),
+		PrivateKey:            planModel.PrivateKey.ValueStringPointer(),
+		TagsOverrideOwnership: planModel.TagsOverrideOwnership.ValueBoolPointer(),
+	}
+	if len(ownershipTagKeys) > 0 {
+		input.OwnershipTagKeys = &ownershipTagKeys
+	}
+
+	createdIntegration, err := r.client.CreateIntegrationGCP(input)
+	if err != nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("Unable to create Google Cloud integration, got error: '%s'", err))
+		return
+	}
+
+	stateModel := newIntegrationGoogleCloudResourceModel(*createdIntegration, planModel)
+
+	tflog.Trace(ctx, "created a Google Cloud integration")
+	resp.Diagnostics.Append(resp.State.Set(ctx, &stateModel)...)
+}
+
+func (r *integrationGoogleCloudResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var stateModel integrationGoogleCloudResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &stateModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	readIntegration, err := r.client.GetIntegration(opslevel.ID(stateModel.Id.ValueString()))
+	if err != nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("Unable to read Google Cloud integration, got error: '%s'", err))
+		return
+	}
+
+	verifiedStateModel := newIntegrationGoogleCloudResourceModel(*readIntegration, stateModel)
+
+	// Save updated data into Terraform state
+	tflog.Trace(ctx, "read a Google Cloud integration")
+	resp.Diagnostics.Append(resp.State.Set(ctx, &verifiedStateModel)...)
+}
+
+func (r *integrationGoogleCloudResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var planModel integrationGoogleCloudResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	input := opslevel.GoogleCloudIntegrationInput{
+		ClientEmail:           planModel.ClientEmail.ValueStringPointer(),
+		Name:                  planModel.Name.ValueStringPointer(),
+		PrivateKey:            planModel.PrivateKey.ValueStringPointer(),
+		TagsOverrideOwnership: planModel.TagsOverrideOwnership.ValueBoolPointer(),
+	}
+	ownershipTagKeys, diags := SetValueToStringSlice(ctx, planModel.OwnershipTagKeys)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	// schema requires at least one ownership tag,
+	if planModel.OwnershipTagKeys.IsNull() {
```

Comment:
The `SetValueToStringSlice` returns an empty string slice already.  This isn't needed.

Reply:
Will change this is in the Azure code too